core/src/main/scala/cats/std/vector.scala

```diff
@@ -68,16 +68,11 @@ trait VectorInstances {
   implicit def eqVector[A](implicit ev: Eq[A]): Eq[Vector[A]] =
     new Eq[Vector[A]] {
       def eqv(x: Vector[A], y: Vector[A]): Boolean = {
-        def loop(xs: Vector[A], ys: Vector[A]): Boolean =
-          xs match {
-            case Seq() => ys.isEmpty
-            case a +: xs =>
-              ys match {
-                case Seq() => false
-                case b +: ys => if (ev.neqv(a, b)) false else loop(xs, ys)
-              }
-          }
-        loop(x, y)
+        @tailrec def loop(from: Int): Boolean =
+          if(from == x.size) true
```

Comment:
Not that my benchmarking is any rigorous, but taking the average of 20 runs in a REPL of the above current state of this PR and this new code that doesn't call `x.size` repeatedly:

```scala
def eqv(x: Vector[A], y: Vector[A]): Boolean = {
  @tailrec def loop(to: Int): Boolean =
    if(to == -1) true
    else ev.eqv(x(to), y(to)) && loop(to - 1)

  (x.size == y.size) && loop(x.size - 1)
}
```

it appears this new version is on average faster by some 21ms (167.95ms vs 146.81ms).

Also, I haven't tried to tweak any optimization flags.